geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/list/LInsertDUnitTest.java

```diff
@@ -98,24 +98,35 @@ public void givenBucketsMoveDuringLInsert_operationsAreNotLost() throws Exceptio
     lpushPerformAndVerify(keys.get(1), elementList2);
     lpushPerformAndVerify(keys.get(2), elementList3);
 
-    Runnable task1 = () -> linsertPerformAndVerify(keys.get(0), BEFORE,
-        jedis.lindex(keys.get(0), 2), insertedValue, numInserts);
-    Runnable task2 = () -> linsertPerformAndVerify(keys.get(1), AFTER,
-        jedis.lindex(keys.get(1), 2), insertedValue, numInserts);
-    Runnable task3 = () -> linsertPerformAndVerify(keys.get(2), AFTER,
-        jedis.lindex(keys.get(2), 2), insertedValue, numInserts);
+    Runnable task1 = () -> {
+      while (continueInserting.get()) {
+        linsertPerformAndVerify(keys.get(0), BEFORE, jedis.lindex(keys.get(0), 2), insertedValue);
+      }
+    };
+    Runnable task2 = () -> {
+      while (continueInserting.get()) {
+        linsertPerformAndVerify(keys.get(1), AFTER, jedis.lindex(keys.get(1), 2), insertedValue);
+      }
+    };
+    Runnable task3 = () -> {
+      while (continueInserting.get()) {
+        linsertPerformAndVerify(keys.get(2), AFTER, jedis.lindex(keys.get(2), 2), insertedValue);
+      }
+    };
 
     Future<Void> future1 = executor.runAsync(task1);
     Future<Void> future2 = executor.runAsync(task2);
     Future<Void> future3 = executor.runAsync(task3);
 
-    for (int i = 0; i < 50 && numInserts.get() > 0; i++) {
+    for (int i = 0; i < 50; i++) {
+      if (i == 20) {
+        continueInserting.set(false);
+      }
```

Comment:
This could be simplified a bit to just have the loop run 20 times, then call `continueInserting.set(false);` afterwards, since we don't care about moving buckets if there are no operations ongoing.